File: cos/cos.olympus/src/main/java/cos/olympus/game/events/Death.java
package cos.olympus.game.events;

import cos.olympus.game.Creature;

public record Death(
        int id,
        int tick,
        Spell spell,
        Creature victim
) {

    public cos.ops.Death toUserOp(int userId) {
        return new cos.ops.Death(id, tick, userId, spell.source().id(), victim.id());
    }

    @Override public String toString() {
        return "Death{" +
                "id=" + id +
                ", tick=" + tick +
                ", victim=" + victim.id() +
                ", spell=" + spell.id() +
                '}';
    }
}

File: cos/cos.olympus/src/main/kotlin/cos/olympus/game/actions/Death.kt
package cos.olympus.game.actions

import cos.olympus.Tsm
import cos.olympus.game.Creature

data class Death(
    override val x: Int,
    override val y: Int,
    override val time: Tsm,
    val victim: Creature,
    val culprit: Creature
) : Action {
    constructor(d: Damage) : this(d.x, d.y, d.time, d.victim, d.culprit)
}
